app/main/menu.ts

```diff
@@ -81,3 +82,49 @@ export function createMenu(window: BrowserWindow) {
     const menu = Menu.buildFromTemplate(template);
     Menu.setApplicationMenu(menu);
 }
+
+export function addContextMenu(window: BrowserWindow) {
+    window.webContents.on('context-menu', (_, params) => {
+        const menu = Menu.buildFromTemplate([
+            { label: 'Cut', role: 'cut', enabled: params.editFlags.canCut },
+            { label: 'Copy', role: 'copy', enabled: params.editFlags.canCopy },
+            {
+                label: 'Paste',
+                role: 'paste',
+                enabled: params.editFlags.canPaste,
+            },
+        ]);
+
+        if (params.dictionarySuggestions.length > 0) {
+            menu.append(new MenuItem({ type: 'separator' }));
+        }
+
+        // Add each spelling suggestion
+        for (const suggestion of params.dictionarySuggestions) {
```

Comment:
How many are suggestions are included? Could it be possible that the list could be too long, making the context menu huge, or is this limited by electron or on OS level?

Reply:
We should check it on each OS, but this is how electron does it in their own documentation:
https://www.electronjs.org/docs/tutorial/spellchecker

Reply:
On linux it does seem to never exceed 5 suggestions.

Reply:
I haven't had it exceed 5 either on macOS (tried 3 times), maybe it's fine then 🙂